src/test/java/org/opensearch/flowframework/bwc/FlowFrameworkBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,189 @@
+        OLD,
+        MIXED,
+        UPGRADED;
+
+        public static ClusterType parse(String value) {
+            switch (value) {
+                case "old_cluster":
+                    return OLD;
+                case "mixed_cluster":
+                    return MIXED;
+                case "upgraded_cluster":
+                    return UPGRADED;
+                default:
+                    throw new AssertionError("unknown cluster type: " + value);
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testBackwardsCompatibility() throws Exception {
+        String uri = getUri();
+        Map<String, Map<String, Object>> responseMap = (Map<String, Map<String, Object>>) getAsMap(uri).get("nodes");
+        for (Map<String, Object> response : responseMap.values()) {
+            List<Map<String, Object>> plugins = (List<Map<String, Object>>) response.get("plugins");
+            Set<Object> pluginNames = plugins.stream().map(map -> map.get("name")).collect(Collectors.toSet());
+            String workflowId = createNoopTemplate();
+            Template t = getTemplate(workflowId);
+            switch (CLUSTER_TYPE) {
+                case OLD:
+                    assertTrue(pluginNames.contains("opensearch-flow-framework"));
+                    // mapping for 2.12 does not include time stamps
+                    assertNull(t.createdTime());
+                    assertNull(t.lastUpdatedTime());
+                    assertNull(t.lastProvisionedTime());
+                    break;
+                case MIXED:
+                    assertTrue(pluginNames.contains("opensearch-flow-framework"));
+                    // Time stamps may or may not be null depending on whether index has been accessed by new version node
+                    // So just test that the template parses
+                    assertNull(t.lastProvisionedTime());
+                    break;
+                case UPGRADED:
+                    assertTrue(pluginNames.contains("opensearch-flow-framework"));
+                    // mapping for 2.13+ includes time stamps
+                    assertNotNull(t.createdTime());
+                    assertEquals(t.createdTime(), t.lastUpdatedTime());
+                    assertNull(t.lastProvisionedTime());
+                    break;
+            }
+            break;
+        }
+    }
+
+    private String getUri() {
+        switch (CLUSTER_TYPE) {
+            case OLD:
+                return "_nodes/" + CLUSTER_NAME + "-0/plugins";
+            case MIXED:
+                String round = System.getProperty("tests.rest.bwcsuite_round");
+                if (round.equals("second")) {
+                    return "_nodes/" + CLUSTER_NAME + "-1/plugins";
+                } else if (round.equals("third")) {
+                    return "_nodes/" + CLUSTER_NAME + "-2/plugins";
+                } else {
+                    return "_nodes/" + CLUSTER_NAME + "-0/plugins";
+                }
```

Comment:
How about?
```suggestion
      int nodeId = round.equals("second") ? 1 : round.equals("third") ? 2 : 0;
      return "_nodes/" + CLUSTER_NAME + "-" + nodeId + "/plugins";
```

Reply:
Another case of "this code was directly copied from other BWC implementations".  I'd rather leave working code alone than try to make it look different than others, making it harder to track/copy bug fixes from other implementations if they ever occur.

Reply:
No harm in improving it. I'll leave that choice to you.

Reply:
Personally I think it’s more readable as is. :)